### PR TITLE
tera-cli: init at 0.2.5

### PR DESCRIPTION
**Pull request by 365tuwe**
## Description of changes

Add tera-cli, a command line utility on top of the tera templating engine. Takes json|yaml|toml as input and can merge ENV in. You may see it as envsubst on steroid. 

https://github.com/chevdor/tera-cli/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/te/tera-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
```
Comment (MikaelFangel) [inline pkgs/by-name/te/tera-cli/package.nix:4]:
```suggestion
  rustPlatform
```

File: pkgs/by-name/te/tera-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "tera-cli";
+  version = "0.2.5";
+
+  src = fetchFromGitHub {
+    owner = "chevdor";
+    repo = pname;
```
Comment (MikaelFangel) [inline pkgs/by-name/te/tera-cli/package.nix:12]:
```suggestion
    repo = "tera-cli";
```

File: pkgs/by-name/te/tera-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    rev = version;
+    hash = "sha256-W+pcVLxOlikwAGvx0twm23GyCMzdqnHY0YBNtcsSB5I=";
+  };
+
+  cargoHash = "sha256-A01mok8KQk1FV8P7E4svdBCW6xqpduHy1XuUcdDFjfc=";
+
+  meta = with lib; {
+    description = "A command line utility written in Rust to render templates from json|toml|yaml && ENV, using the tera templating engine";
```
Comment (MikaelFangel) [inline pkgs/by-name/te/tera-cli/package.nix:20]:
```suggestion
    description = "A command line utility to render templates from json|toml|yaml && ENV, using the tera templating engine";
```

**Review by MikaelFangel — COMMENTED** [review]
Thanks for your contribution to nixpkgs. 

Your PR should only contain two commits:
```
maintainers: add _365tuwe
tera-cli: init at 0.2.5
```

and the pr tittle should be: tera-cli: init at 0.2.5

File: pkgs/by-name/te/tera-cli/package.nix (hunk 3 above)
Comment (365tuwe) [reply pkgs/by-name/te/tera-cli/package.nix:20]: This is the project's original description from https://github.com/chevdor/tera-cli/blob/9f845e7ed47efcc434e66098f279da08146bf1e2/Cargo.toml#L3 - is there a reason not to use this?

Comment (MikaelFangel) [reply pkgs/by-name/te/tera-cli/package.nix:20]: `meta.description` is for a brief description of what the package does, and there I don't see what language a package is written in a good fit. (: 

[https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/meta.chapter.md#description-var-meta-description](https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/meta.chapter.md#description-var-meta-description)

File: pkgs/by-name/te/tera-cli/package.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  rustPlatform
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "tera-cli";
+  version = "0.2.5";
+
+  src = fetchFromGitHub {
+    owner = "chevdor";
+    repo = "tera-cli";
+    rev = version;
```
Comment (MikaelFangel) [inline pkgs/by-name/te/tera-cli/package.nix:13]:
```suggestion
    rev = "v${version}";
```

Comment (MikaelFangel) [reply pkgs/by-name/te/tera-cli/package.nix:13]: Else you get 404 error not found.

**Comment by MikaelFangel** [comment]
I get the following error when running the example from the repo.

```
❯ ./result/bin/tera --template template.tera data.json
Error: 
   0: failed to create context
   1: failed to append file to context
   2: failed to parse JSON
   3: expected value at line 1 column 1

Location:
   src/wrapped_context.rs:41

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  <empty backtrace>

Run with COLORBT_SHOW_HIDDEN=1 environment variable to disable frame filtering.
Run with RUST_BACKTRACE=full to include source snippets.

❯ cat data.json                                       
.data.json:
{
    "title": "Demo",
    "users": [
        {
            "username": "Alice",
            "url": "http://example.org/alice",
            "fav_colors": ["red", "green", "yellow"]
        },
        {
            "username": "Bob",
            "url": "http://example.org/bob",
            "fav_colors": ["orange"]
        }
    ]
}

❯ cat template.tera     
.template.tmpl
<title>{% block title %} {{title}} {% endblock title %}</title>

<ul>
{% for user in users -%}
    <li><a href="{{ user.url }}">{{ user.username }}
    {{ user.username }} likes {% for color in user.fav_colors -%}{{ color }} {% endfor %}
    </a></li>
{% endfor %}
</ul>

```

**Comment by 365tuwe** [comment]
> I get the following error when running the example from the repo.
> 
> ```
> ❯ ./result/bin/tera --template template.tera data.json
> Error: 
>    0: failed to create context
>    1: failed to append file to context
>    2: failed to parse JSON
>    3: expected value at line 1 column 1
> 
> Location:
>    src/wrapped_context.rs:41
> 
>   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
>   <empty backtrace>
> 
> Run with COLORBT_SHOW_HIDDEN=1 environment variable to disable frame filtering.
> Run with RUST_BACKTRACE=full to include source snippets.
> 
> ❯ cat data.json                                       
> .data.json:
> {
>     "title": "Demo",
>     "users": [
>         {
>             "username": "Alice",
>             "url": "http://example.org/alice",
>             "fav_colors": ["red", "green", "yellow"]
>         },
>         {
>             "username": "Bob",
>             "url": "http://example.org/bob",
>             "fav_colors": ["orange"]
>         }
>     ]
> }
> 
> ❯ cat template.tera     
> .template.tmpl
> <title>{% block title %} {{title}} {% endblock title %}</title>
> 
> <ul>
> {% for user in users -%}
>     <li><a href="{{ user.url }}">{{ user.username }}
>     {{ user.username }} likes {% for color in user.fav_colors -%}{{ color }} {% endfor %}
>     </a></li>
> {% endfor %}
> </ul>
> ```

This is an issue with the example, the first line of the json file makes it invalid. Remove `.data.json:` and you're good.

**Review by MikaelFangel — APPROVED** [review]
Thanks. Then it looks good to me, at least. (:

**Review by nikstur — APPROVED** [review]

**Comment by nikstur** [comment]
Thank you for your first contribution and welcome to Nixpkgs!

**Pull request merged**